index.ts

```diff
@@ -61,3 +70,54 @@ client.on("messageCreate", (messages: Message) => {
     loggerService.log(error);
   }
 });
+
+client.on("message", async (message) => {
+  const directMessage = new DirectMessage(message, client, channelResolver);
+  const validationCheck = await directMessage.validate();
+  if (validationCheck) {
+    directMessage.messageApprove();
+  }
```

Comment:
Como referi num comment acima, se mudarmos o naming da função podemos inclusive melhorar a legibilidade deste check e até usar `early returns` para a validação. (se quisermos até podemos dar logo embed do await no if.

```suggestion
  const isValid = await directMessage.isValid();
  if (isValid) {
    return;
  }
  
  directMessage.messageApprove();
```

Reply:
feito